### PR TITLE
Add types for the gitignore-parser package

### PR DESCRIPTION
**Pull request by anuccio1**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/gitignore-parser/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    maybe(input: string): boolean;
+};
+
+/**
+ * Parse the given `.gitignore` content and return an array
+ * containing two further arrays - positives and negatives.
+ * Each of these two arrays in turn contains two regexps, one
+ * strict and one for 'maybe'.
+ */
+export function parse(content: string): RegExp[][];
```
Comment (microsoftly) [inline types/gitignore-parser/index.d.ts:28]: I think we can be more explicit with the typings here. We know that a 2D array of size 2X2 of RegExp is being returned.

so
```typescript
export type GitignoreParseResultSet = [RegExp, RegExp]; // [strict, maybe]

export function parse(content: string): [GitignoreParseResultSet, GitignoreParseResultSet]; // [positive, negative]
```

Comment (anuccio1) [reply types/gitignore-parser/index.d.ts:28]: great suggestion

File: types/gitignore-parser/gitignore-parser-tests.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import gitignore = require('gitignore-parser');
+
+const gitCompile = gitignore.compile('.git');
+
+const isAccept: boolean = gitCompile.accepts('test');
+const isDeny: boolean = gitCompile.maybe('test');
+const isMaybe: boolean = gitCompile.denies('test');
+
+const parseVal = gitignore.parse('.git');
```
Comment (microsoftly) [inline types/gitignore-parser/gitignore-parser-tests.ts]: adding a typedef to parseVal like `isAccept`, `isDeny`, etc ... to ensure typechecks will catch typedef changes down the line

**Review by microsoftly — APPROVED** [review]
LGTM

**Pull request merged**